mvrx/src/main/kotlin/com/airbnb/mvrx/lifecycleAwareLazy.kt

```diff
@@ -29,9 +34,18 @@ class lifecycleAwareLazy<out T>(private val owner: LifecycleOwner, initializer:
     private val lock = this
 
     init {
-        // owner.lifecycle.addObserver must be invoked on main thread otherwise addObserver will throw an IllegalStateException.
-        // createUnsafe disables the main thread check.
-        LifecycleRegistry.createUnsafe(owner).addObserver(object : DefaultLifecycleObserver {
+        if (isMainThread()) {
+            initializeOnCreatedOrGreater(owner)
+        } else {
+            Handler(Looper.getMainLooper()).post {
+                if (owner.lifecycle.currentState < Lifecycle.State.CREATED || isInitialized()) return@post
+                initializeOnCreatedOrGreater(owner)
+            }
+        }
+    }
+
+    private fun initializeOnCreatedOrGreater(owner: LifecycleOwner) {
+        owner.lifecycle.addObserver(object : DefaultLifecycleObserver {
```

Comment:
as an optimization this could check the state and if it is created or greater then do the initialization immediately, instead of going through the work to create, add, and then remove an observer

Reply:
Oh yer. Sure, that sounds like a plan. Updated PR